pkg/test/e2e/defaultappcatalog/application_test.go

```diff
@@ -360,3 +363,17 @@ func createUserCluster(
 
 	return clusterClient, cleanup, log, err
 }
+
+// makeValuesBlockWithTemplating takes a string of values and returns a string in order to be used for templating.
+func makeValuesBlockWithTemplating(name string, values string) string {
+	var customLine string
+	if strings.Contains(defaultValuesBlock, "null") || defaultValuesBlock == "" {
+		defaultValuesBlock = ""
+		customLine = fmt.Sprintf("global:\n  customVar: %s\n", customVarTemplate)
+	} else {
+		// Ensure proper newline separation if appending
+		customLine = fmt.Sprintf("\nglobal:\n  customVar: %s\n", customVarTemplate)
+	}
+
+	return values + customLine
```

Comment:
I would strongly prefer if we use yaml.Marshal&Unmarshal here. I am surprised that the current solution works, but as soon as any of the default values contain a global block, like this

```yaml
#default values
global:
  something: something
otherkey:
  bla: blub
```

we would produce the following yaml, which is invalid

```yaml
#default values
global:
  something: something
otherkey:
  bla: blub
global:
  customVar: ...
```